### PR TITLE
bugfix: could not serialize and deserialize in java.sql.Blob, java.sql.Clob

### PR DESCRIPTION
**Pull request by l81893521**
### Ⅰ. Describe what this PR did
Fix could not serialize and deserialize in java.sql.Blob, java.sql.Clob

### Ⅱ. Does this pull request fix one issue?
#1636

### CODE REVIEW

File: rm-datasource/src/main/java/io/seata/rm/datasource/undo/parser/JacksonUndoLogParser.java [hunk 1]
```diff
@@ -158,4 +187,84 @@ public Timestamp deserialize(JsonParser p, DeserializationContext ctxt) {
...
+
+        @Override
+        public SerialBlob deserialize(JsonParser p, DeserializationContext ctxt)
+            throws IOException {
+            try {
+                return new SerialBlob(p.getBinaryValue());
+            } catch (SQLException e) {
+                LOGGER.error("deserialize java.sql.Blob error : {}", e.getMessage(), e);
+            }
+            LOGGER.error("deserialize java.sql.Blob type error.");
```
[the remark this answers is not in the document]
Comment (l81893521) [reply rm-datasource/src/main/java/io/seata/rm/datasource/undo/parser/JacksonUndoLogParser.java]: ok

File: rm-datasource/src/main/java/io/seata/rm/datasource/sql/struct/TableRecords.java [hunk 2]
```diff
@@ -182,7 +188,16 @@ public static TableRecords buildRecords(TableMeta tmeta, ResultSet resultSet) th
                     field.setKeyType(KeyType.PrimaryKey);
                 }
                 field.setType(col.getDataType());
-                field.setValue(resultSet.getObject(i));
+                if(col.getDataType() == JDBCType.BLOB.getVendorTypeNumber()){
```
[the remark this answers is not in the document]
Comment (l81893521) [reply rm-datasource/src/main/java/io/seata/rm/datasource/sql/struct/TableRecords.java]: ok

File: rm-datasource/src/main/java/io/seata/rm/datasource/undo/parser/JacksonUndoLogParser.java [hunk 3]
```diff
@@ -158,4 +187,84 @@ public Timestamp deserialize(JsonParser p, DeserializationContext ctxt) {
...
+        @Override
+        public SerialClob deserialize(JsonParser p, DeserializationContext ctxt)
+            throws IOException {
+            try {
+                return new SerialClob(p.getValueAsString().toCharArray());
+
+            } catch (SQLException e) {
+                LOGGER.error("deserialize java.sql.Clob error : {}", e.getMessage(), e);
+            }
+            LOGGER.error("deserialize java.sql.Clob type error.");
```
[the remark this answers is not in the document]
Comment (l81893521) [reply rm-datasource/src/main/java/io/seata/rm/datasource/undo/parser/JacksonUndoLogParser.java]: ok

**Comment by l81893521** [comment]
can not rollback

**Comment by l81893521** [comment]
fix can not roll back

File: rm-datasource/src/main/java/io/seata/rm/datasource/sql/struct/TableRecords.java [hunk 4]
```diff
@@ -182,7 +188,16 @@ public static TableRecords buildRecords(TableMeta tmeta, ResultSet resultSet) th
                     field.setKeyType(KeyType.PrimaryKey);
                 }
                 field.setType(col.getDataType());
-                field.setValue(resultSet.getObject(i));
+                if (col.getDataType() == JDBCType.BLOB.getVendorTypeNumber()) {
```
Comment (slievrly) [inline rm-datasource/src/main/java/io/seata/rm/datasource/sql/struct/TableRecords.java]: in fact，just oracle can run in this code. if db is mysql and the col type=blob, col.getDataType() will return LONGVARBINARY（-4）.

**Pull request merged**